components/cli/pkg/commands/image/test.go

```diff
@@ -221,221 +222,160 @@ func startTestCellInstance(imageDir string, instanceName string, runningNode *de
-			}
+	} else {
+		err = util.CleanOrCreateDir(testsRoot)
+		if err != nil {
+			return fmt.Errorf("error occurred while creating %s, %v", testsRoot, err)
 		}
+	}
 
-		if exePath == "" {
-			mntImageDir := strings.Replace(imageDir, util.UserHomeDir(), "/home/cellery", 1)
-			err = CreateBallerinaConf(string(iName), verboseMode, mntImageDir, string(dependencyLinksJson), envVars, testsRoot)
-			if err != nil {
-				return fmt.Errorf("error occurred while creating the ballerina.conf, %s", err)
-			}
-			err = RunDockerTelepresenceTests(incell, cmd, testsRoot, imageDir, instanceName)
-			if err != nil {
-				StopTelepresence(telepresenceYamlPath)
-				return fmt.Errorf("error occured while running tests in ballerina docker image, %s", err)
-			}
-		} else {
-			err = CreateBallerinaConf(string(iName), verboseMode, imageDir, string(dependencyLinksJson), envVars, testsRoot)
-			if err != nil {
-				return fmt.Errorf("error occurred while creating the ballerina.conf, %s", err)
-			}
-			cmd = &exec.Cmd{}
-			cmd.Env = os.Environ()
-			cmd.Env = append(cmd.Env, constants.CelleryImageDirEnvVar+"="+imageDir)
-
-			//if incell {
-			//	cmdArgs = append(cmdArgs, "--groups", "incell")
-			//} else {
-			//	cmdArgs = append(cmdArgs, "--disable-groups", "incell")
-			//}
-			err = RunTelepresenceTests(incell, cmd, exePath, imageDir, instanceName, debug)
-			if err != nil {
-				StopTelepresence(telepresenceYamlPath)
-				return fmt.Errorf("error occured while running tests, %s", err)
-			}
-		}
-	} else {
-		if exePath == "" {
-			return fmt.Errorf("Ballerina should be installed to debug tests.")
-		} else {
-			balProj, err := filepath.Abs(projLocation)
-			if err != nil {
-				return fmt.Errorf("error occurred while retrieving absolute path of project, %s", err)
-			}
-			PromtConfirmation(assumeYes, balProj)
-			err = CreateBallerinaConf(string(iName), verboseMode, imageDir, string(dependencyLinksJson), envVars, balProj)
-			if err != nil {
-				return fmt.Errorf("error occurred while creating the ballerina.conf, %s", err)
-			}
-			cmd = &exec.Cmd{}
-			cmd.Env = os.Environ()
-			cmd.Env = append(cmd.Env, constants.CelleryImageDirEnvVar+"="+imageDir)
-
-			err = RunTelepresenceTests(incell, cmd, exePath, imageDir, instanceName, debug)
-			if err != nil {
-				StopTelepresence(telepresenceYamlPath)
-				return fmt.Errorf("error occured while running tests, %s", err)
-			}
+	// Copy artifacts for running tests
+	if isBallerinaProject {
+		fileCopyError := util.CopyFile(balTomlPath, filepath.Join(testsRoot, constants.BallerinaToml))
+		if fileCopyError != nil {
+			return fmt.Errorf("error occurred while copying %s, %v", constants.BallerinaToml, err)
 		}
 	}
-	StopTelepresence(telepresenceYamlPath)
+	fileCopyError := util.CopyDir(testsPath, filepath.Join(balModule, constants.ZipTests))
+	if fileCopyError != nil {
+		return fmt.Errorf("error occurred while copying tests folder, %s", err)
```

Comment:
err should be fileCopyError, and use %v to return error object.

Reply:
Fixed